server/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -559,42 +551,30 @@ protected final void doClose() {
 
     @Override
     protected final void doStop() {
-        final CountDownLatch latch = new CountDownLatch(1);
-        // make sure we run it on another thread than a possible IO handler thread
-        assert threadPool.generic().isShutdown() == false : "Must stop transport before terminating underlying threadpool";
```

Comment:
Should we keep this? Sounds like the sort of invariant we should check, we're probably assuming it in other places.

Reply:
Yea good point :) lets keep it.